pkg/plugins/manager.go

```diff
@@ -0,0 +1,170 @@
+package plugins
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var (
+	errPluginGroupNotFound = fmt.Errorf("unknown plugin group")
+	errPluginTypeNotFound  = fmt.Errorf("unknown plugin type")
+)
+
+// PluginType specifies the type of related plugins
+// For example, "diff" is a type of plugins that know how to perform a diff.
+type PluginType string
+
+// PluginName specifies the name of a plugin
+// For example, "delta" is a plugin.
+type PluginName string
+
+// PluginIdentity identifies the plugin implementation.
+type PluginIdentity struct {
+	Impl               plugin.Plugin
+	ImplName           string
+	HandshakeConfig    plugin.HandshakeConfig
+	ExecutableLocation string
+}
+type pluginTypeConfigMap map[PluginName]plugin.ClientConfig
+
+// PluginTypeNameMap example:
```

Comment:
Thought we have two sets of configurations - one specify the plugins+config. The second maps the operation to a plugin.